nucypher/blockchain/eth/actors.py

```diff
@@ -1171,6 +1174,47 @@ def missing_commitments(self) -> int:
         missing = self.staking_agent.get_missing_commitments(checksum_address=staker_address)
         return missing
 
+    def _verify_policy_payment_by_policy_id(self, policy_id: str) -> bool:
+        from nucypher.policy.policies import Policy
+        arrangements = self.policy_agent.fetch_policy_arrangements(policy_id=policy_id[:Policy.POLICY_ID_LENGTH])
+        for arrangement in arrangements:
+            if self.checksum_address == arrangement.node:
+                return True
+        else:
+            raise self.UnpaidPolicy(f"{policy_id} is unpaid.")
+
+    def _verify_policy_payment_by_txhash(self, txhash: bytes, timeout: int) -> bool:
+        # TODO: We'd love for this to be impossible to reduce the risk of collusion.  #1274
+
+        blockchain = self.policy_agent.blockchain
+        try:
+            blockchain.client.wait_for_receipt(txhash, timeout=timeout)
+            transaction = blockchain.client.w3.eth.getTransaction(txhash)
+        except TimeExhausted:
+            raise self.UnpaidPolicy
+
+        transaction_data = blockchain.client.parse_transaction_data(transaction)
+        _signature, parameters = self.policy_agent.contract.decode_function_input(transaction_data)
+        try:
+            # Get all of the arrangements and verify that we'll be paid.
+            arranged_addresses = parameters['_nodes']
+        except KeyError:
+            raise ValueError(f'Txhash {txhash.hex()} is not a policy creation transaction.')
+
+        this_node_has_been_arranged = self.checksum_address in arranged_addresses
+        if this_node_has_been_arranged:
+            return True
+        else:
+            raise self.UnpaidPolicy
+
+    def verify_policy_payment(self, timeout: int, policy_id: str = None, txhash: bytes = None) -> bool:
```

Comment:
May be a silly question, but I see here that we check that a payment transaction exists, and that's it. Are we not interested in *how much* we were paid? Is it handled somewhere else?

Reply:
That's a great question. Fortunately, the smart contract guarantees that a _successful_ TX to `createPolicy()` includes a correct payment. 

...wait a minute...."_successful_ TX"....what about failed TXs?! We're not checking the status of the TX! 🙈 

It should be possible to send a TX calling `createPolicy()` that fails for whatever reason (e.g., not enough funds) and try to pass this failed TX as a "proof of payment", 

It was not a silly question after all, @fjarri.